src/libaktualizr/package_manager/dockerappmanager.cc

```diff
@@ -0,0 +1,99 @@
+#include "dockerappmanager.h"
+
+#include <sstream>
+
+struct DockerApp {
+  DockerApp(const std::string app_name, const PackageConfig &config)
+      : name(std::move(app_name)),
+        app_root(std::move(config.docker_apps_root / app_name)),
+        app_params(std::move(config.docker_app_params)),
+        app_bin(std::move(config.docker_app_bin)),
+        compose_bin(std::move(config.docker_compose_bin)) {}
+
+  bool render(const std::string &app_content) {
+    auto bin = boost::filesystem::canonical(app_bin).string();
+    Utils::writeFile(app_root / (name + ".dockerapp"), app_content);
+    std::string cmd("cd " + app_root.string() + " && " + bin + " app render " + name);
+    if (!app_params.empty()) {
+      cmd += " -f " + app_params.string();
+    }
+    std::string yaml;
+    if (Utils::shell(cmd, &yaml, true) != 0) {
+      LOG_ERROR << "Unable to run " << cmd << " output:\n" << yaml;
+      return false;
+    }
+    Utils::writeFile(app_root / "docker-compose.yml", yaml);
+    return true;
+  }
+
+  bool start() {
+    auto bin = boost::filesystem::canonical(compose_bin).string();
+    std::string cmd("cd " + app_root.string() + " && " + bin + " up --remove-orphans -d");
+    if (std::system(cmd.c_str()) != 0) {
```

Comment:
Good question. The "start" command can take a while to start so std::system will print out progress as its made. So basically its a little nicer this way when watching things run. Not sure that's enough to justify doing things differently than everywhere else in the project though.

Reply:
If there's a good reason, it's no problem. It is also interesting to consider a version of `Utils::shell` that piped output directly to the logger. For now, it's probably fine, but I'd suggest leaving a comment to explain what you just explained here so that it doesn't get changed by an overzealous refactorer in the future.